pkg/collect/certificates.go

```diff
@@ -0,0 +1,221 @@
+
+	output := NewResult()
+	results := []CertCollection{}
+
+	// collect certificates from secrets
+	for secretName, namespaces := range c.Collector.Secrets {
+		for _, namespace := range namespaces {
+
+			secretCollections := secretCertCollector(secretName, namespace, c.Client)
+			results = append(results, secretCollections...)
+		}
+	}
+
+	/*
+		// collect certificates from configMaps
+		for configMapName, namespaces := range c.Collector.ConfigMaps {
+			for _, namespace := range namespaces {
+				configMapCollections := configMapCertCollector(configMapName, namespace, c.Client)
+				results = append(results, configMapCollections...)
+			}
+		}
+	*/
+
+	certsJson, _ := json.MarshalIndent(results, "", "\t")
+
+	filePath := "certificates/certificates.json"
+
+	output.SaveResult(c.BundlePath, filePath, bytes.NewBuffer(certsJson))
+
+	return output, nil
+}
+
+// configmap certificate collector function will
+func configMapCertCollector(configMapName string, namespace string, client kubernetes.Interface) []CertCollection {
+
+	results := []CertCollection{}
+
+	//listOptions := metav1.ListOptions{}
+	//configMaps, _ := client.CoreV1().ConfigMaps(namespace).List(context.Background(), listOptions)
+
+	getOptions := metav1.GetOptions{}
+
+	// Collect from configMaps
+	configMap, _ := client.CoreV1().ConfigMaps(namespace).Get(context.Background(), configMapName, getOptions)
+
+	trackErrors := []string{}
+
+	collection := []ParsedCertificate{}
+
+	//Collect from configMap
+	source := &CertificateSource{
+		ConfigMapName: configMap.Name,
+		Namespace:     configMap.Namespace,
+	}
+	for certName, c := range configMap.Data {
+
+		certs := []byte(c)
+
+		certInfo, _ := CertParser(certName, certs)
+
+		collection = append(collection, certInfo...)
+	}
+	results = append(results, CertCollection{
+		Source:           source,
+		Errors:           trackErrors,
+		CertificateChain: collection,
+	})
+
+	return results
+}
+
+// secret certificate collector function
+func secretCertCollector(secretName string, namespace string, client kubernetes.Interface) []CertCollection {
+
+	results := []CertCollection{}
+
+	//listOptions := metav1.ListOptions{}
+	//secrets, _ := client.CoreV1().Secrets(namespace).List(context.Background(), listOptions)
+
+	// TODO: Handle RBAC errors. Not to be worked on yet
+	getOptions := metav1.GetOptions{}
+	// Collect from secrets
+	secret, _ := client.CoreV1().Secrets(namespace).Get(context.Background(), secretName, getOptions)
+
+	trackErrors := []string{}
+
+	collection := []ParsedCertificate{}
+
+	// Collect from secret
+	source := &CertificateSource{
+		SecretName: secret.Name,
+		Namespace:  secret.Namespace,
+	}
+
+	for certName, certs := range secret.Data {
+		certInfo, _ := CertParser(certName, certs)
+
+		collection = append(collection, certInfo...)
+
+	}
+	results = append(results, CertCollection{
+		Source:           source,
+		Errors:           trackErrors,
```

Comment:
For trackErrors field, it contains the error from CertParser in design, right?

Reply:
Hi @DexterYan Thank you for reviewing!  A TODO for tomorrow is to shore up the error collections; the CertParser for sure need a bit of touch with regards to trackErrors; so I agree with our asessement.